libfwupdplugin/fu-archive.c

```diff
@@ -234,6 +257,137 @@ fu_archive_load(FuArchive *self, GBytes *blob, FuArchiveFlags flags, GError **er
 #endif
 }
 
+static GByteArray *
+fu_archive_save(FuArchive *self,
+		const gchar *compress,
```

Comment:
maybe `compression_type` or `compress_format` perhaps? We've used compress in other places as a boolean.

Reply:
I am creating an enum

Reply:
With what contents? Isn't the underlying archive format pretty free text?

Reply:
If you push what you have now, I've got a few other ideas how the write stuff could work too.

Reply:
well, I did not want to rely on their strings